client/state/selectors/is-recurring-payments-stats-nudge-visible.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+import { get, isArray } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	isCurrentPlanPaid,
+	isJetpackSite,
+	isJetpackMinimumVersion,
+	getSite,
+} from 'state/sites/selectors';
+import { userCan } from 'lib/site/utils';
+import isRecurringPaymentsStatsNudgeDismissed from 'state/selectors/is-recurring-payments-stats-nudge-dismissed';
+
+/**
+ * Returns true if the Recurring Payments nudge should be visible in stats
+ * Context: https://wordpress.com/read/blogs/122326716/posts/8496#comment-3782
+ * It should be visible for sites that:
+ * - are on a paid plan (feature availability)
+ * - avarage more than a 1000 uniques per month
+ *
+ * This selector makes a point of not requesting any new data, so tries to reuse stats being already available in state tree
+ * Sometimes they are not 100% accurate, but this is a nudge and it's good enough for our purposes.
+ *
+ * @param  {object}  state  Global state tree
+ * @param  {string}  siteId The Site ID
+ * @returns {boolean} True if we should show the nudge
+ */
+export default function isRecurringPaymentsStatsNudgeVisible( state, siteId ) {
+	if ( ! isCurrentPlanPaid( state, siteId ) ) {
+		return false;
+	}
+	// We don't want to show notice if JP is too old
+	if (
+		isJetpackSite( state, siteId ) &&
+		isJetpackMinimumVersion( state, siteId, '7.4' ) === false
+	) {
+		return false;
+	}
+	// We don't want to show the notice to the user that can't use recurring payments
+	if ( ! userCan( 'manage_options', getSite( state, siteId ) ) ) {
+		return false;
+	}
+
+	if ( isRecurringPaymentsStatsNudgeDismissed( state, siteId ) ) {
```

Comment:
Same comment here about using date math in a selector. Is there a way we can do the math part inside the places where it's used instead?

Reply:
We have the same pattern in:
https://github.com/Automattic/wp-calypso/blob/f2c9d99b07e14ea938829e9b2e5b31b6fb388a90/client/state/selectors/is-gsuite-stats-nudge-dismissed.js#L47
https://github.com/Automattic/wp-calypso/blob/f2c9d99b07e14ea938829e9b2e5b31b6fb388a90/client/state/selectors/is-google-my-business-stats-nudge-dismissed.js#L51

And other selectors for these nudges